blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/processors/BatchWriter.java

```diff
@@ -108,49 +104,36 @@ else if (m instanceof PreaggregatedMetric)
                         } finally {
                             dirtyTimerCtx.stop();
                         }
-                        
+
                         return true;
                     } catch (Exception ex) {
                         getLogger().error(ex.getMessage(), ex);
-                        successfullyPersisted.set(false);
+                        getLogger().warn("Did not persist all metrics successfully for batch " + batchId);
                         return false;
                     } finally {
                         singleBatchWriteCtx.stop();
-                        shortLatch.countDown();
                         bufferedMetrics.dec(batch.size());
-                        
-                        if (System.currentTimeMillis() - writeStartTime > timeout.toMillis()) {
-                            writeTimedOut.set(true);
-                        }
-                        done();
-                    }
-                    
-                    
-                }
-                
-                private void done() {
-                    if (shortLatch.getCount() == 0) {
-                        getLogger().debug("Successfully persisted all metrics for batch " + batchId);
-                        actualWriteCtx.stop();
 
-                        if (writeTimedOut.get()) {
+                        if (System.currentTimeMillis() - writeStartTime > timeout.toMillis()) {
                             exceededScribeProcessingTime.mark();
                             getLogger().error("Exceeded timeout " + timeout.toString() + " before persisting " +
                                     "all metrics for batch " + batchId);
                         }
-    
-                        if (!successfullyPersisted.get()) {
-                            getLogger().warn("Did not persist all metrics successfully for batch " + batchId);
-                        }
                     }
                 }
-                
             }); 
             
             resultFutures.add(futureBatchResult);
         }
         
-        return Futures.allAsList(resultFutures);
+        ListenableFuture<List<Boolean>> finalFuture = Futures.allAsList(resultFutures);
+        finalFuture.addListener(new Runnable() {
+            @Override
+            public void run() {
+                actualWriteCtx.stop();
+            }
+        }, MoreExecutors.sameThreadExecutor());
```

Comment:
sameThreadExecutor() has been deprecated in favor of directExecutor(): http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/util/concurrent/MoreExecutors.html#sameThreadExecutor()

Otherwise this looks good.